packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -482,7 +506,10 @@ export function renderWithHooks<Props, SecondArg>(
   // This check uses currentHook so that it works the same in DEV and prod bundles.
   // hookTypesDev could catch more cases (e.g. context) but only in DEV bundles.
   const didRenderTooFewHooks =
-    currentHook !== null && currentHook.next !== null;
+    (currentHook !== null && currentHook.next !== null) ||
+    (current !== null &&
+      current.memoizedState !== null &&
+      workInProgress.memoizedState == null);
```

Comment:
So I think this is the risky part that makes it throw. So maybe let's hold off with this or put it behind a flag?

Reply:
> Or maybe an alternate fix is to add warnInvalidHookAccess to HooksDispatcherOnMountWithHookTypesInDEV.

Yes that's how I would do it

Reply:
Specifically in the `mountHookTypesDev` function, to make it symmetrical with `updateHookTypesDev`